code/game/machinery/syndicatebomb.dm

```diff
@@ -292,6 +326,127 @@
+		if(S.Uses)
+			for(var/obj/item/weapon/reagent_containers/glass/G in beakers)
+				G.reagents.trans_to(S, G.reagents.total_volume)
+
+			if(S && S.reagents && S.reagents.total_volume)
+				reactants += S.reagents
+
+	if(!chem_splash(get_turf(src), spread_range, reactants, temp_boost))
+		playsound(loc, 'sound/items/Screwdriver2.ogg', 50, 1)
+		return // The Explosion didn't do anything. No need to log, or disappear.
+
+	if(adminlog)
+		message_admins(adminlog)
+		log_game(adminlog)
+
+	playsound(loc, 'sound/effects/bamf.ogg', 75, 1, 5)
+
+	if(loc && istype(loc,/obj/machinery/syndicatebomb/))
+		qdel(loc)
+	qdel(src)
+
+/obj/item/weapon/bombcore/chemical/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/crowbar) && beakers.len > 0)
+		playsound(loc, 'sound/items/Crowbar.ogg', 50, 1)
+		for (var/obj/item/B in beakers)
+			B.loc = get_turf(src)
+			beakers -= B
+		return
+	else if(istype(I, /obj/item/weapon/reagent_containers/glass/beaker) || istype(I, /obj/item/weapon/reagent_containers/glass/bottle))
+		if(beakers.len < max_beakers)
+			if(!user.drop_item())
+				return
+			beakers += I
+			user << "<span class='notice'>You load [src] with [I].</span>"
+			I.loc = src
+		else
+			user << "<span class='warning'>The [I] wont fit! The [src] can only hold up to [max_beakers] containers.</span>"
+			return
+	..()
+
+/obj/item/weapon/bombcore/chemical/CheckParts()
+	// Using different grenade casings, causes the payload to have different properties.
+	for(var/obj/item/weapon/grenade/chem_grenade/G in src)
+
+		if(istype(G, /obj/item/weapon/grenade/chem_grenade/large))
+			var/obj/item/weapon/grenade/chem_grenade/large/LG = G
+			max_beakers += 1 // Adding two large grenades only allows for a maximum of 6 beakers.
+			spread_range += 2 // Extra range, reduced density.
+			temp_boost += 50 // maximum of +150K blast using only large beakers. Not enough to self ignite.
+			for(var/obj/item/slime_extract/S in LG.beakers) // And slime cores.
+				if(beakers.len < max_beakers)
+					beakers += S
+					S.loc = src
+				else
+					S.loc = get_turf(src)
+
+		if(istype(G, /obj/item/weapon/grenade/chem_grenade/cryo))
+			spread_range -= 1 // Reduced range, but increased density.
+			temp_boost -= 100 // minimum of -150K blast.
+
+		if(istype(G, /obj/item/weapon/grenade/chem_grenade/pyro))
+			temp_boost += 150 // maximum of +350K blast, which is enough to self ignite. Which means a self igniting bomb can't take advantage of other grenade casing properties. Sorry?
+
+		if(istype(G, /obj/item/weapon/grenade/chem_grenade/adv_release))
+			time_release += 25 // A typical bomb, using basic beakers, will explode over 2-4 seconds. Using two will make the reaction last for less time, but it will be more dangerous overall.
+
+		for(var/obj/item/weapon/reagent_containers/glass/B in G)
+			if(beakers.len < max_beakers)
+				beakers += B
+				B.loc = src
+			else
+				B.loc = get_turf(src)
+
+		qdel(G)
+	return
```

Comment:
You do not have to include the "return" command at the end of a proc, as it will do so automatically.